### PR TITLE
Migrate CircleCI unit test workflow to Github  (contrib)

### PR DESCRIPTION
**Pull request by shovnik**
## Which problem is this PR solving?
- This addresses [issue](https://github.com/open-telemetry/opentelemetry-js/issues/1259#event-4038172785) by migrating the only existing opentelemetry-js-contrib workflow from CircleCI to GitHub Actions (config.yml) 

## Short description of the changes
- Migrate node-unit-tests and browser-unit-tests contained in config.yml file to GitHub Actions
- Change lint workflow trigger to be on pull_request and push as well to avoid overlapping triggers
- Minor unrelated lint fix so that the lint test stops failing for every PR on the contrib repo

## Migration Plan
I suggest having CircleCI and GitHub Action jobs run in parallel for a few weeks. After the GitHub Action jobs are running fine for a week or so and then remove the CircleCI workflows from config.yml

cc - @alolita

### CODE REVIEW

File: plugins/node/opentelemetry-instrumentation-graphql/src/types.ts [hunk 1]
```diff
@@ -63,9 +63,7 @@ export interface GraphQLInstrumentationConfig {
 /**
  * Merged and parsed config of default instrumentation config and GraphQL
  */
-export type GraphQLInstrumentationParsedConfig = Required<
-  GraphQLInstrumentationConfig
-> &
+export type GraphQLInstrumentationParsedConfig = Required<GraphQLInstrumentationConfig> &
```
Comment (AzfaarQureshi) [inline plugins/node/opentelemetry-instrumentation-graphql/src/types.ts:66]: this change should be checked out because its unrelated to the GHA migration

**Review by AzfaarQureshi — APPROVED** [review]
LGTM, just 1 minor comment

**Review by alolita — APPROVED** [review]
lgtm

**Comment by shovnik** [comment]
Filed upstream

**Pull request closed without merging**